stdlib/universe/hourSelection.go

```diff
@@ -0,0 +1,173 @@
+}
+
+func createHourSelectionOpSpec(args flux.Arguments, a *flux.Administration) (flux.OperationSpec, error) {
+	if err := a.AddParentFromArgs(args); err != nil {
+		return nil, err
+	}
+
+	spec := new(HourSelectionOpSpec)
+
+	if start, ok, err := args.GetInt("start"); err != nil {
+		return nil, err
+	} else if ok {
+		spec.Start = start
+	} else {
+		spec.Start = int64(0)
+	}
+
+	if stop, ok, err := args.GetInt("stop"); err != nil {
+		return nil, err
+	} else if ok {
+		spec.Stop = stop
+	} else {
+		spec.Stop = int64(0)
+	}
+
+	return spec, nil
+}
+
+func newHourSelectionOp() flux.OperationSpec {
+	return new(HourSelectionOpSpec)
+}
+
+func (s *HourSelectionOpSpec) Kind() flux.OperationKind {
+	return HourSelectionKind
+}
+
+type HourSelectionProcedureSpec struct {
+	plan.DefaultCost
+	Start int64
+	Stop  int64
+}
+
+func newHourSelectionProcedure(qs flux.OperationSpec, pa plan.Administration) (plan.ProcedureSpec, error) {
+	spec, ok := qs.(*HourSelectionOpSpec)
+	if !ok {
+		return nil, fmt.Errorf("invalid spec type %T", qs)
+	}
+
+	return &HourSelectionProcedureSpec{
+		Start: spec.Start,
+		Stop:  spec.Stop,
+	}, nil
+}
+
+func (s *HourSelectionProcedureSpec) Kind() plan.ProcedureKind {
+	return HourSelectionKind
+}
+
+func (s *HourSelectionProcedureSpec) Copy() plan.ProcedureSpec {
+	ns := new(HourSelectionProcedureSpec)
+
+	*ns = *s
+
+	return ns
+}
+
+func (s *HourSelectionProcedureSpec) TriggerSpec() plan.TriggerSpec {
+	return plan.NarrowTransformationTriggerSpec{}
+}
+
+func createHourSelectionTransformation(id execute.DatasetID, mode execute.AccumulationMode, spec plan.ProcedureSpec, a execute.Administration) (execute.Transformation, execute.Dataset, error) {
+	s, ok := spec.(*HourSelectionProcedureSpec)
+	if !ok {
+		return nil, nil, fmt.Errorf("invalid spec type %T", spec)
+	}
+	cache := execute.NewTableBuilderCache(a.Allocator())
+	d := execute.NewDataset(id, mode, cache)
+	t := NewHourSelectionTransformation(d, cache, s)
+	return t, d, nil
+}
+
+type hourSelectionTransformation struct {
+	d     execute.Dataset
+	cache execute.TableBuilderCache
+
+	start int64
+	stop  int64
+}
+
+func NewHourSelectionTransformation(d execute.Dataset, cache execute.TableBuilderCache, spec *HourSelectionProcedureSpec) *hourSelectionTransformation {
+	return &hourSelectionTransformation{
+		d:     d,
+		cache: cache,
+		start: spec.Start,
+		stop:  spec.Stop,
+	}
+}
+
+func (t *hourSelectionTransformation) RetractTable(id execute.DatasetID, key flux.GroupKey) error {
+	return t.d.RetractTable(key)
+}
+
+func (t *hourSelectionTransformation) Process(id execute.DatasetID, tbl flux.Table) error {
+	builder, created := t.cache.TableBuilder(tbl.Key())
+	if !created {
+		return fmt.Errorf("hour selection found duplicate table with key: %v", tbl.Key())
+	}
+	if err := execute.AddTableCols(tbl, builder); err != nil {
+		return err
+	}
+
+	colIdx := execute.ColIdx("_time", builder.Cols())
```

Comment:
I think you can run in this check on `tbl` instead of builder and directly use that index below (for `j`) without  re-searching the column index.